Calling/ClientApp/src/components/MediaGallery.tsx

```diff
@@ -96,29 +84,39 @@ export default (props: MediaGalleryProps): JSX.Element => {
     return remoteParticipantsMediaGalleryItems;
   };
 
-  const numberOfColumns = calculateNumberOfColumns(props.remoteParticipants, Constants.DOMINANT_PARTICIPANTS_COUNT);
-  if (numberOfColumns !== gridCol) setGridCol(numberOfColumns);
-  const numberOfRows = calculateNumberOfRows(props.remoteParticipants, Constants.DOMINANT_PARTICIPANTS_COUNT);
-  if (numberOfRows !== gridRow) setGridRow(numberOfRows);
+  // determine number of rows/columns to add to the grid
+  const numberStreamsToRender = useMemo(() => clamp(props.remoteParticipants.length, 0, numRemoteParticipantsToRender), [props.remoteParticipants.length])
```

Comment:
nit: useCallback ?

Reply:
I believe we just want to save the value based on the number of participants. If we useCallback we will need to re-run the function it will just not have a new instance for this "component". If we useMemo we just get to use the same value until the number of participants change again. In this case I think it makes sense  to useMemo